src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -102,17 +104,33 @@ public static void finalizeRegion(final AssemblyRegion region,
         final List<GATKRead> readsToUse = region.getReads().stream()
                 // TODO unclipping soft clips may introduce bases that aren't in the extended region if the unclipped bases
                 // TODO include a deletion w.r.t. the reference.  We must remove kmers that occur before the reference haplotype start
-                .map(read -> skipSoftClips || ! ReadUtils.hasWellDefinedFragmentSize(read) ?
+                .map(read -> dontUseSoftClippedBases || ! ReadUtils.hasWellDefinedFragmentSize(read) ?
                     ReadClipper.hardClipSoftClippedBases(read) : ReadClipper.revertSoftClippedBases(read))
-                .map(read -> ReadClipper.hardClipLowQualEnds(read, minTailQualityToUse))
+                .map(read -> softClipLowQualityEnds ? ReadClipper.softClipLowQualEnds(read, minTailQualityToUse) :
+                        ReadClipper.hardClipLowQualEnds(read, minTailQualityToUse))
                 .filter(read -> read.getStart() <= read.getEnd())
                 .map(read -> read.isUnmapped() ? read : ReadClipper.hardClipAdaptorSequence(read))
                 .filter(read ->  !read.isEmpty() && read.getCigar().getReadLength() > 0)
                 .map(read -> ReadClipper.hardClipToRegion(read, region.getPaddedSpan().getStart(), region.getPaddedSpan().getEnd() ))
                 .filter(read -> read.getStart() <= read.getEnd() && read.getLength() > 0 && read.overlaps(region.getPaddedSpan()))
-                .sorted(new ReadCoordinateComparator(readsHeader)) // TODO: sort may be unnecessary here
                 .collect(Collectors.toList());
 
+        // This will probably have to change...
```

Comment:
There is some context for this. Everything got scrunched into the streaming operation but at each stage where we clip reads above we end up making copies and throwing away the transient attributes and there is no easy way to thread the original read through without refactoring all the methods. I don't want to change the clipping operations and I also don't want to wrap the reads into some object that stores them and I don't necessarily want to revert to the old codepath for this method that wasn't in the streaming operation (since we have made several changes to this method since refacotring and I can't be sure I did it correctly). I can't use existing attributes for GATKRead since those don't let me shove an entire read object into them. I went with an... expedient solution at the time. 

 Its worth noting that this corresponded to the mismatch with DRAGEN that sounded like unintended behavior where they were using the original (unmodified/non-realigined/non-clipped) reads for selecting genotype overlaps... It sounds like this doesn't make enough of a difference to be necessary for DRAGEN-GATK so my preference is to pull all of this original tag storing code and the argument alltogether rather than try to force a hackey solution onto the engine.  @kachulis @droazen